File: mentorship/sprints/week#2/sprint-2/1B.cpp
#include <bits/stdc++.h>
using namespace std;
int n;

int str2int(string s){
	int total = 0;
	for (int i = 0; i < (int)s.length(); ++i){
		total *= 10;
		total += s[i] - '0';
	}
	return total;
}

int main(){
	scanf("%d", &n);
	while(n--){
		string word;
		cin>>word;
		string col = "", row = "";

		bool flag = false;
		int idx;
		for(int i = 1; i < (int)word.length()-1; i++)
			if(isdigit(word[i]) && isalpha(word[i+1])){
				idx = i+1;
				flag = true;
				break;
			}

		if(word[0] == 'R' && flag){
			row = word.substr(1, idx-1);
			col = word.substr(idx+1);
			int column = str2int(col);
			
			string res = "";
			while(column != 0){
				column -= 1;
				int location = column % 26;
				res = char(location + 'A') + res;
				column /= 26;
			}
			cout<<res<<row<<endl;
		}
		else{
			for (int i = 0; i < (int)word.length(); ++i)
				if(isdigit(word[i])){
					row = word.substr(i);
					col = word.substr(0, i);
					break;
				}
			string res = "R" + row + "C";
			int column = 0;

			for(char c : col){
				column *= 26;
				column += (c - 'A') + 1;
			}

			cout<<res<<column<<endl;
		}
	}
	return 0;
}

File: mentorship/sprints/week#2/sprint-2/1B.py
n = int(input())

for _ in range(n):
	word = input()
	col = ""
	row = ""
	

	flag = False
	for i in range(1, len(word)-1):
		if(word[i].isdigit() and word[i+1].isalpha()):
			flag = True
			break
	
	if(word[0] == 'R' and flag):
		for i in range(1, len(word)):
			if(word[i].isalpha()):
				row = word[1:i]
				col = word[i+1:]


		col = int(col)
		res = ""
		while(col != 0):
			col -= 1
			loc = col % 26
			res = chr(loc + ord('A')) + res
			col //= 26
		print(res+row)

	else:
		for i in range(len(word)):
			if(word[i].isdigit()):
				col = word[:i]
				row = word[i:]
				break

		res = "R"+row+"C"

		column = 0
		for c in col:
			column *= 26
			column += (ord(c) - ord('A') + 1)
		res += str(column)
		print(res)

